apps/daimo-mobile/src/view/screen/OnboardingScreen.tsx

```diff
@@ -36,35 +38,93 @@ import {
   TextLight,
 } from "../shared/text";
 
-export default function OnboardingScreen() {
-  const [page, setPage] = useState(1);
-  const [accountFlowChoice, setAccountFlowChoice] = useState<
-    "create" | "existing"
-  >("create");
-  const next = useCallback(() => setPage(page + 1), [page]);
-  const onFlowChoice = useCallback(
-    (choice: "create" | "existing") => {
-      setAccountFlowChoice(choice);
-      setPage(page + 1);
+type OnboardPage =
+  | "intro"
+  | "create"
+  | "existing"
+  | "new-allow-notifications"
+  | "existing-allow-notifications"
+  | "new-loading";
+
+export default function OnboardingScreen({
+  onOnboardingComplete,
+}: {
+  onOnboardingComplete: () => void;
+}) {
+  const [page, setPage] = useState<OnboardPage>("intro");
+
+  const next = useCallback(
+    (choice?: "create" | "existing") => {
+      setPage(
+        (function () {
+          switch (page) {
+            case "intro":
+              return assertNotNull(choice);
+            case "create":
+              return "new-allow-notifications";
+            case "existing":
+              return "existing-allow-notifications";
+            case "new-allow-notifications":
+              return "new-loading";
+            case "existing-allow-notifications":
+            case "new-loading":
+              onOnboardingComplete();
+              return page;
+            default:
+              throw new Error(`unreachable ${page}`);
+          }
+        })()
+      );
     },
     [page]
   );
 
-  // TODO: add back buttons?
+  // TODO: add back buttons on create and existing pages
+  const prev = useCallback(() => {}, []);
```

Comment:
easy to implement now that we've refactored `page`

copy above, edit switch cases ^

Reply:
will leave for later, requires implementing a "header" and intercepting android back buttons as well.